fe/fe-core/src/main/java/com/starrocks/master/MasterImpl.java

```diff
@@ -1261,6 +1304,23 @@ public TCommitRemoteTxnResponse commitRemoteTxn(TCommitRemoteTxnRequest request)
     public TAbortRemoteTxnResponse abortRemoteTxn(TAbortRemoteTxnRequest request) throws TException {
         TAbortRemoteTxnResponse response = new TAbortRemoteTxnResponse();
         Catalog catalog = Catalog.getCurrentCatalog();
+
+        // if current node is follower, forward it to leader
+        if (!catalog.isMaster()) {
+            try {
+                TNetworkAddress addr = masterAddr();
+                LOG.info("abortRemoteTxn as follower, forward it to master. txn_id: {}, master: {}",
+                         request.getTxn_id(), addr.toString());
+                FrontendService.Client client = ClientPool.frontendPool.borrowObject(addr, 1000);
+                response = client.abortRemoteTxn(request);
+                ClientPool.frontendPool.returnObject(addr, client);
```

Comment:
Better to return object in `finally {}` and also need to invalidate object if the rpc failed.